jupyter notebook with all of python code [here](https://github.com/hades-k/Kazakh_pop_Exome/blob/main/1000_gen_v2.ipynb)
______

Starting from all_phase3_exome files: 

#### Check again that the file contains only SNPs, and create plink 1.9 files

```bash
plink2 --pfile all_phase3_exome --snps-only just-acgt \
	--make-bed --out 1000_gen_snps
```
2504 samples (1270 females, 1233 males, 1 ambiguous; 2497 founders) loaded from
all_phase3_exome.psam.
148456 out of 156719 variants loaded from all_phase3_exome.pvar.
2 categorical phenotypes loaded.
148456 variants remaining after main filters.

#### Check missingness and MAF and filter 

```bash
plink --bfile 1000_gen_snps --missing --out qc/missingness
plink --bfile 1000_gen_snps --freq --out qc/freqs
```
Ambiguous sex ID written to qc/freqs.nosex 

Removing the ambigous sex ID, and filtering:

```bash 
plink --bfile 1000_gen_snps --remove qc/freqs.nosex --geno 0.02 \
	--mind 0.02 --maf 0.01 --make-bed --out qc/1000_gen_filtered
```

148456 variants loaded from .bim file.
2504 people (1233 males, 1270 females, 1 ambiguous) loaded from .fam.
--remove: 2503 people remaining.
0 people removed due to missing genotype data (--mind).
Before main variant filters, 2496 founders and 7 nonfounders present.
Total genotyping rate in remaining samples is exactly 1.
0 variants removed due to missing genotype data (--geno).
18499 variants removed due to minor allele threshold(s)
129957 variants and 2503 people pass filters and QC.

#### Extract variants present in Kazakh set 

```bash
awk '{print $2}' 8_multialleles_removed/gold_hetero_filtered_cleaned_8.bim \
	> kazakh_snps.txt
plink --bfile qc/1000_gen_filtered --extract kazakh_snps.txt \
	--make-bed --out matched/1000_gen_matched
```
129957 variants loaded from .bim file.
2503 people (1233 males, 1270 females) loaded from .fam.
--extract: 129957 variants remaining.

Doing the same for Kazakhs

```bash
awk '{print $2}' matched/1000_gen_matched.bim > matched_snps.txt
plink --bfile 8_multialleles_removed/gold_hetero_filtered_cleaned_8 \
	--extract matched_snps.txt --make-bed \
	--out matched/kazakh_matched
```
175105 variants loaded from .bim file.
121 people (64 males, 57 females) loaded from .fam.
--extract: 129957 variants remaining.

#### Matching the genome build

```bash 
awk '{print $2, $4}' matched/kazakh_matched.bim > buildmap.txt
plink --bfile matched/1000_gen_matched --update-map buildmap.txt \
	--make-bed --out coord_sync/1000_gen_buildsync
```
129957 variants loaded from .bim file.
2503 people (1233 males, 1270 females) loaded from .fam.
--update-map: 129957 values updated.
Warning: Base-pair positions are now unsorted!

Sorting the positions:

```bash
plink2 --bfile coord_sync/1000_gen_buildsync --sort-vars --make-pgen \
	--out coord_sync/1000_gen_sorted_pgen
plink2 --pfile coord_sync/1000_gen_sorted_pgen --make-bed \
	--out coord_sync/1000_gen_sorted
```

#### Set reference genome

```bash
awk '{print $2, $5}' coord_sync/1000_gen_sorted.bim > ref_alleles.txt
plink --bfile matched/kazakh_matched --reference-allele ref_alleles.txt  \
	--make-bed --out set_ref/kazakh_refset
```

Total genotyping rate is 0.997159.
Warning: Impossible A1 allele assignment for variant rs3790723.
Warning: Impossible A1 allele assignment for variant rs1047883.
Warning: Impossible A1 allele assignment for variant rs13153811.
Warning: Impossible A1 allele assignment for variant rs12807209.
Warning: Impossible A1 allele assignment for variant rs80058880.
Warning: Impossible A1 allele assignment for variant rs73305632.
--a1-allele: 129957 assignments attempted, 129951 made.

#### Removing impossible A1 alleles

Since we have only 6 impossible alleles, we could just remove them without losing much information 

```bash
grep 'Impossible A1 allele' set_ref/kazakh_refset.log | awk '{print $(NF)}' \
	> impossible_a1_snps.txt
```
The rs ids have dots in the end, remove them manually 

```bash
plink --bfile set_ref/kazakh_refset --exclude impossible_a1_snps.txt \
	--make-bed --out re-matched/kazakh_noimpossible
```
129957 variants loaded from .bim file.
121 people (64 males, 57 females) loaded from .fam.
--exclude: 129951 variants remaining.

Exclude them from the 1000 genomes files as well

```bash
awk '{print $2}' re-matched/kazakh_noimpossible.bim > snp_list.txt
plink --bfile coord_sync/1000_gen_sorted --extract snp_list.txt \
	--make-bed --out re-matched/1000_genomes_noimpossible
```

129957 variants loaded from .bim file.
2503 people (1233 males, 1270 females) loaded from .fam.
--extract: 129951 variants remaining.

#### Strand flipping

Check for differences:

```bash
awk '{print $2, $5, $6}' kazakh_noimpossible.bim | sort > kazakh_vars.txt
awk '{print $2, $5, $6}' 1000_genomes_noimpossible.bim | sort > 1000gen_vars.txt
comm -3 kazakh_vars.txt 1000gen_vars.txt > differences.txt
wc -l differences.txt
```
13234 differences.txt

```bash 
head differences.txt
```

Output:
```
	rs10009558 C T
rs10009558 T C
rs1001238 C T
	rs1001238 T C
rs10018049 C T
	rs10018049 T C
	rs10018746 C T
rs10018746 T C
	rs10018837 A G
rs10018837 G A
```

Flip the alleles in Kazakh dataset 

```bash
awk '{print $1}' re-matched/differences.txt | sort -u > to_flip.txt
plink --bfile re-matched/kazakh_noimpossible --flip to_flip.txt \
	--make-bed --out flip/kazakh_flipped
```
129951 variants loaded from .bim file.
121 people (64 males, 57 females) loaded from .fam.
--flip: 6617 SNPs flipped.

#### Final exclusion 

After checking for differences again, we got 11426 differences_2.txt, and looking at the head:

```
rs10009558 A G
	rs10009558 C T
rs1001238 G A
	rs1001238 T C
rs10018049 G A
	rs10018049 T C
rs10018746 A G
	rs10018746 C T
	rs10018837 A G
rs10018837 C T
```
Removing them: 
```bash
awk '{print $1}' differences_2.txt | sort -u > final_exclude.txt
plink --bfile flip/kazakh_flipped --exclude final_exclude.txt \
	--make-bed --out final_exclude/kazakh_final
plink --bfile re-matched/1000_genomes_noimpossible \
	--exclude final_exclude.txt --make-bed \
	--out final_exclude/1000_gen_final
```

#### Subsetting 

Subset to keep only the populations of interest:

```
Bengali
British
Dai Chinese
Finnish
Gujarati
Han Chinese
Iberian
Japanese
Kinh Vietnamese
Punjabi
Tamil
Toscani
```

```bash
grep 'BEB' filtered_igsr_samples.tsv | shuf -n 10 \
	> 8_subset/Bengali_subset.tsv
cat 8_subset/*_subset.tsv | awk '{print "0 "$1}' > ids_to_keep.txt
plink --bfile 7_final_exclude/1000_gen_final --keep ids_to_keep.txt \
	--make-bed --out 8_subset/1000gen_subset
```
124238 variants and 120 people pass filters and QC.

Note: use population codes instead of population names to avoid mishaps such as Han Chinese vs Southern Han Chinese

#### Merge 

```bash
plink --bfile 7_final_exclude/kazakh_final \
	--bmerge 8_subset/1000gen_subset --make-bed \
	--out 9_merge/merged
```
121 people loaded from 7_final_exclude/kazakh_final.fam.
120 people to be merged from 8_subset/1000gen_subset.fam.
Of these, 120 are new, while 0 are present in the base dataset.

124238 markers loaded from 7_final_exclude/kazakh_final.bim.
124238 markers to be merged from 8_subset/1000gen_subset.bim.
Of these, 0 are new, while 124238 are present in the base dataset.

241 people (118 males, 123 females) loaded from .fam.
Before main variant filters, 241 founders and 0 nonfounders present.
Total genotyping rate is 0.998575.
124238 variants and 241 people pass filters and QC.


#### PCA and analysis 

```bash
plink --bfile 9_merge/merged --pca 10 --out 10_pca/pca_res
```

Comparing the eigenvalues, and caluclating the formula $$Var \ explained_i = \frac { \lambda_i }{ \sum_{j=1}^{n} \lambda_j}$$ we get the following: 

```
Variance explained by each PC:
PC1: 34.0%
PC2: 13.2%
PC3: 8.2%
PC4: 7.6%
PC5: 6.6%
PC6: 6.3%
PC7: 6.2%
PC8: 6.1%
PC9: 6.0%
PC10: 5.8%
```
And plotting first 5 PC's we get this:

![3x3_pca](https://github.com/user-attachments/assets/b7f73d14-767d-427d-ab00-0dc2abd7a7a5)

- PC 1 shows the East Asian <-> Kazakh <-> European gradient (South Asian close to european??)
- PC 2 mainly separate the South Asian population from the rest
- PC 3 refines some substructure in East Asian population; 3 Kazakh samples are separated vastly from the rest
- PC's 4 and 5 show some unknown substructure in Kazakh population. Regional? Noise?


#### Remove otliers 

Focusing on the fist 3 principal components, compute the Euclidean distance of each Kazakh individual from the Kazakh population centroid in the 3D PCA space. Marking the ones that have a z-score > 2 (2+ standard deviations from the mean), we get this:

| #   | IID   | Population | dist_from_kz_center | kz_dist_zscore |
| --- | ---   | ---        | ----                | ----           | 
| 121 | WE002 | Kazakh     | 0.343428            | 6.247338       |
| 190 | WE091 | Kazakh     | 0.319514            | 5.774943       |
| 191 | WE092 | Kazakh     | 0.336037            | 6.101342       |

Could be unknown admixture or sequencing errors. Save them to outliers.txt and exclude them from the plink files 

```bash
plink --bfile 7_final_exclude/kazakh_final --remove outliers.txt \
	--make-bed --out 11_remove_outliers/kazakh_no_outliers
```
--remove: 118 people remaining.
124238 variants and 118 people pass filters and QC.

Merge and PCA

```bash
plink --bfile 11_remove_outliers/kazakh_no_outliers \
	--bmerge 8_subset/1000gen_subset --make-bed \
	--out 12_merge_no_outliers_pca/merged_no_outliers

plink --bfile 12_merge_no_outliers_pca/merged_no_outliers \
	--pca 10 --out 12_merge_no_outliers_pca/pca_no_outliers
```

<img src = 'https://github.com/hades-k/Kazakh_pop_Exome/blob/main/pca_plots/3x3_pca_no_outliers.png'>

- PC 1 shows the East Asian <-> Kazakh <-> European gradient (South Asian close to european)
- PC 2 mainly separate the South Asian population from the rest
- PC 3 refines significant substructure in East Asian population; refines some inner substructure in Kazakh population (unknown, regional?); refines substructure in European population
- PC 4 onward do not contain information relevant for further study. Could be noise.

Plotting the values in a 3D space, we can get the following graph:

<img src = 'https://github.com/hades-k/Kazakh_pop_Exome/blob/main/pca_plots/3d_pca.png'>

#### Clustering 

Performing K-means clustering using scikit-learn python package. Most informative K was chosen to be 4 (graphs [here](https://github.com/hades-k/Kazakh_pop_Exome/blob/main/pca_plots/kmeans_k_diff_pcs_3.png) and [here](https://github.com/hades-k/Kazakh_pop_Exome/blob/main/pca_plots/kmeans_4_diff_pcs.png) ). Focusing on the top 3 pricipal components, we get the following clusters: 

<img src = 'https://github.com/hades-k/Kazakh_pop_Exome/blob/main/pca_plots/clustering.png'>

Locating the cluster that has most of the individuals from Kazakh population (cluster 1 contains 100%), measured the euclidian distances from the center of cluster 1 to the rest:

| Cluster   | Distance to C1 |
| ---       | ---            |
| Cluster 0 | 0.1731         |
| Cluster 2 | 0.1739         |
| Cluster 3 | 0.1788         |

#### ADMIXTURE

First, find the optimal K value: how many ancestral populations are we interested in. To do this, run different values of K and record CV error. For example:

```bash
admixture --cv ../12_merge_no_outliers_pca/merged_no_outliers.bed 5
```
Cross-Validation error rates:

| K | CV error rate |
| - | ---           |
| 2 | 0.44369	    |
| 3 | 0.44419	    |
| 4 | 0.44601 	    |
| 5 | 0.45226	    |


Looking at the output and the Cross-Validation errors (min at k=2, increasing with every k), decided to try again with a bigger subset from other populations. Testing the hypothesis that with the current subsets of 10 per population, admixture is biased heavily towards Kazakh population. 

Make a new subset with 30 individuals per population (same populations as previously), giving total of 120 per superpopulation (EAS, SAS, EUR). Merge them with kazakh_no_outliers. Admixture attempt #2.

```bash
admixture --cv big_subset/merged.bed 2 | tee big_subset_res/log2.out
```
This time we have the follwing Cross-Validation error rates: 

| K | CV error rate |
| - | ---           |
| 2 | 0.43293	    |
| 3 | 0.42749	    |
| 4 | 0.42688 	    |
| 5 | 0.42889	    |
| 6 | 0.43036	    |

We can see that overall, the error rates are smaller, thereforee we can be more confident in the results. We also have a clear min here, at K=4.

![output](https://github.com/user-attachments/assets/944bd7ae-4831-44d7-805e-eacee42a0364)

_____

Tools used:
- PLINK v1.90p
- PLINK v2.00a5.12
- ADMIXTURE v1.3.0 

Python packages used:
- pandas v2.2.3
- matplotlib v3.10.0
- numpy v2.2.0
- scikit-learn v1.6.1
- plotly 6.0.1n 
- scipy 1.15.1
